OpenImis.RestApi/Security/IMISJwtSecurityTokenHandler.cs

```diff
@@ -69,11 +77,15 @@ public ClaimsPrincipal ValidateToken(string securityToken, TokenValidationParame
             var tokenS = handler.ReadToken(securityToken) as JwtSecurityToken;
             int userId = Convert.ToInt32(tokenS.Claims.Where(w => w.Type == "UserId").Select(x => x.Value).FirstOrDefault());
 
-            UserData user = _imisModules.GetLoginModule().GetLoginLogic().GetById(userId);
+            string apiVersion = _httpContextAccessor.HttpContext.Request.Headers.Where(x => x.Key == "api-version").Select(s => s.Value).FirstOrDefault();
+
+            UserData user;
+            
+            if (apiVersion == null || apiVersion.StartsWith(_lastApiVersion.ToString())) user = (UserData)_imisModulesV2.GetLoginModule().GetLoginLogic().GetById(userId);
```

Comment:
Why we need a variable _lastApiVersion if we know that this is for V2? 
What's happening when we have 3 or 4 versions of API (here we have only 2)?
Could we create the imisModules type based on a string here (i.e. "ModulesV"+_requiredApiVersion|_lastApiVersion+".IImisModules") and then make the call imisModules.GetLoginModule().GetLoginLogic().GetById(userId) using reflection?

Reply:
I would leave it as it is. When we have a finished version 2, we can finally come back and improve it. I will set the version 2 now rigidly